src/AzureRMTools.ts

```diff
@@ -418,8 +433,7 @@ export class AzureRMTools {
             // Code editor. The events will be registered again when the next
             // deployment template is opened.
             if (Object.keys(this._deploymentTemplates).length === 0 && this._deploymentTemplateFileSubscriptions) {
```

Comment:
nit: Since you're checking `this._deploymentTemplateFileSubscriptions` inside `unhookDeploymentFileEvents`, I would prefer to remove that check here

Reply:
I'll consider for later.